pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentConfig.java

```diff
@@ -30,7 +30,7 @@
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class SegmentConfig {
-  private static final int DEFAULT_MAX_NUM_RECORDS_PER_SEGMENT = 5_000_000;
+  public static final int DEFAULT_MAX_NUM_RECORDS_PER_SEGMENT = 5_000_000;
```

Comment:
1M maybe a better default num?

Reply:
1M records per segment is usually too small. Here I'd prefer keeping the existing behavior of 5M per segment.